File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/StringAttribute.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;

/**
 * Represents an attribute with a string value.
 */
public class StringAttribute extends Attribute<String> {
  /**
   * Creates a new StringAttribute instance with the provided key and value.
   * @param id
   * @param value
   */
  public StringAttribute(String id, String value) {
    super(id, value);
  }

  public static StringAttribute fromProto(IdentityEntitiesV2.StringAttributeV2 res) {
    return new StringAttribute(res.getId(), res.getValue());
  }

  public IdentityEntitiesV2.StringAttributeV2 toProto() {
    return IdentityEntitiesV2.StringAttributeV2.newBuilder()
        .setId(this.id)
        .setValue(this.value)
        .build();
  }
}

File: languages/php/Bloock/Entity/IdentityV2/StringAttribute.php
<?php

namespace Bloock\Entity\IdentityV2;

class StringAttribute extends Attribute
{
    public static function fromProto(\Bloock\StringAttributeV2 $res): StringAttribute
    {
        return new StringAttribute($res->getId(), $res->getValue());
    }

    public function toProto(): \Bloock\StringAttributeV2
    {
        $p = new \Bloock\StringAttributeV2();
        $p->setId($this->id);
        $p->setValue($this->value);
        return $p;
    }
}